### PR TITLE
Add parameterfindable and other improvements + fixes

### PR DESCRIPTION
**Pull request by siemensikkema**

### CODE REVIEW

File: Sources/App/Models/Migrations/AppUser.Create.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                .field(.name, .string, .required)
+                .field(.hashedPassword, .string, .required)
+                .field(.shouldResetPassword, .bool, .required)
+                .field(.createdAt, .datetime)
+                .field(.updatedAt, .datetime)
+                .unique(on: .id, .email)
+                .create()
+                .flatMap {
+                    database.createIndex(for: .email, onTable: schema)
+                }
```
Comment (heidipuk) [inline Sources/App/Models/Migrations/AppUser.Create.swift:21]: I tend to use `.transform(to: database.createIndex(for: .email, onTable: schema))` and would love to hear you input on the two approaches 🤔

**Review by heidipuk — APPROVED** [review]
looks great 😄

File: Sources/App/Models/Migrations/AppUser.Create.swift (hunk 1 above)
Comment (siemensikkema) [reply Sources/App/Models/Migrations/AppUser.Create.swift:21]: @heidipuk that function takes an existing future and thus the work has already started. In order to ensure that the second task follows the first you have to use flatMap like this.

Comment (heidipuk) [reply Sources/App/Models/Migrations/AppUser.Create.swift:21]: arhh, yes, okey... I was under the impression that a transform would always follow the task it was chained to and hence it was safe to use. But sounds like it was more luck that it didn't fail for me then 😅

**Pull request merged**